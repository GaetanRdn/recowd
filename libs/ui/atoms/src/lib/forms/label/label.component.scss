@mixin rc-label($config) {
  $typography: map-get(map-get($config, typography), label);

  rc-label {
    @each $key in font-size, font-family, font-weight, line-height {
      #{$key}: map-get($typography, $key);
    }

    color: inherit;
    padding: 1rem 1.6rem 0 1.6rem;
  }

  :is(.rc-required) rc-label::after {
    content: ' *';
  }
}
