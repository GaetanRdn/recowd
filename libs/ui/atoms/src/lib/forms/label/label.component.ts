import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  selector: 'rc-label',
  standalone: true,
  template: ` <ng-content></ng-content>`,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class LabelComponent {}
