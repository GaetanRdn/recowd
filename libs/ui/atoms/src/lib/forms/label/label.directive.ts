import { Directive } from '@angular/core';

@Directive({
  selector: 'rc-label, [rcLabel]',
  standalone: true,
})
export class LabelDirective {}
