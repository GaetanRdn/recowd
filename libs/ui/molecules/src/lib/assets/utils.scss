@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
