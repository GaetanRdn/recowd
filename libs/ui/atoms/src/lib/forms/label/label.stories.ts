import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { BADGE } from '@geometricpanda/storybook-addon-badges';
import { LabelComponent } from './label.component';

export default {
  title: 'atoms/forms/label',
  component: LabelComponent,
  parameters: {
    badges: [BADGE.STABLE],
    jest: 'label.component',
  },
  decorators: [
    moduleMetadata({
      imports: [LabelComponent],
    }),
  ],
} as Meta<LabelComponent>;

const template: Story<LabelComponent> = () => ({
  template: `<rc-label>A label</rc-label>`,
});

export const Default = template.bind({});
