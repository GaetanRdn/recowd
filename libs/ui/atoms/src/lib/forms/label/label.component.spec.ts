import { TestBed, waitForAsync } from '@angular/core/testing';
import { Component } from '@angular/core';
import { TemplateLookup } from '@recowd/test/utils';
import { LabelModule } from './label.component';

describe('LabelComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() =>
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [LabelModule],
    }).compileComponents()));

  beforeEach(() => (templateLookup = new TemplateLookup(HostComponent)));

  it.each(['Name', 'First name'])(
    'should create with label %s',
    (label: string) => {
      templateLookup.hostComponent.label = label;
      templateLookup.detectChanges();
      expect(templateLookup.firstChildElement).toMatchSnapshot();
    }
  );
});

@Component({
  template: ` <rc-label>{{ label }}</rc-label>`,
})
export class HostComponent {
  public label!: string;
}
