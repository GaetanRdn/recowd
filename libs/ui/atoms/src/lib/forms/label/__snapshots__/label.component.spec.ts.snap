// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelComponent should create with label First name 1`] = `
<rc-label>
  First name
</rc-label>
`;

exports[`LabelComponent should create with label Name 1`] = `
<rc-label>
  Name
</rc-label>
`;
